src/mbgl/style/expression/within.cpp

```diff
@@ -0,0 +1,139 @@
+    //  wn = the winding number (=0 only when point is outside)
+    int wn = 0;
+    for (auto poly : polys) {
+        auto size = poly.size();
+        auto i = size;
+        // loop through every edge of the polygon
+        for (i = 0; i < size - 1; ++i) {
+            if (poly[i].y <= point.y) {
+                if (poly[i + 1].y > point.y) { // upward horizontal crossing from point to edge E(poly[i], poly[i+1])
+                    if (isLeft(poly[i], poly[i + 1], point) > 0) {
+                        ++wn;
+                    }
+                }
+            } else {
+                if (poly[i + 1].y <= point.y) { // downward crossing
+                    if (isLeft(poly[i], poly[i + 1], point) < 0) {
+                        --wn;
+                    }
+                }
+            }
+        }
+        if (wn != 0) {
+            return true;
+        }
+    }
+    return wn != 0;
+}
+
+bool pointsWithinPoly(const mbgl::GeometryTileFeature& feature,
+                      const mbgl::CanonicalTileID& canonical,
+                      const mbgl::GeoJSON& geoJson) {
+    bool result = false;
+    geoJson.match(
+        [&](const mapbox::geometry::geometry<double>& geometrySet) {
+            geometrySet.match(
+                [&](const mapbox::geometry::polygon<double>& polygons) {
+                    convertGeometry(feature, canonical)
+                        .match(
+                            [&](const mapbox::geometry::point<double>& point) {
+                                result = pointWithinPoly(point, polygons);
+                            },
+                            [&](const mapbox::geometry::multi_point<double>& points) {
+                                for (const auto& p : points) {
+                                    result = pointWithinPoly(p, polygons);
+                                    if (!result) {
+                                        return;
+                                    }
+                                }
+                            },
+                            [&](const auto&) { return; });
+                },
+                [&](const auto&) { return; });
+        },
+        [&](const auto&) { return; });
+
+    return result;
+}
+
+mbgl::optional<mbgl::GeoJSON> parseValue(const mbgl::style::conversion::Convertible& value_,
+                                         mbgl::style::expression::ParsingContext& ctx) {
+    if (isObject(value_)) {
+        const auto geometryType = objectMember(value_, "type");
+        if (geometryType) {
+            const auto type = toString(*geometryType);
+            if (type && *type == "Polygon") {
+                mbgl::style::conversion::Error error;
+                auto geojson = toGeoJSON(value_, error);
+                if (geojson && error.message.empty()) {
+                    return geojson;
+                }
+                ctx.error(error.message);
```

Comment:
ctx.error on line 91 will overwrite this ctx.error

Reply:
ctx.error() will push back new error messages